src/content/docs/en/reference/routing-reference.mdx

```diff
@@ -0,0 +1,292 @@
+import Since from '~/components/Since.astro';
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';
+import ReadMore from '~/components/ReadMore.astro';
+
+There is no separate routing configuration in Astro. Every [supported page file](/en/basics/astro-pages/#supported-page-files) located within the special `src/pages/` directory creates a single page route, or multiple page routes dynamically when the file name contains a [parameter](#params).
+
+
+## `getStaticPaths()`
+
+**Type:** `(options: GetStaticPathsOptions) => Promise<GetStaticPathsResult> | GetStaticPathsResult`
+
+If a page uses dynamic parameters in the filename, that component will need to export a `getStaticPaths()` function.
+
+This function is required because Astro is a static site builder. That means that your entire site is built ahead of time. If Astro doesn't know how to generate a page at build time, your users won't see it when they visit your site.
+
+```astro
+---
+export async function getStaticPaths() {
+  return [
+    { params: { /* required */ }, props: { /* optional */ } },
+    { params: { ... } },
+    { params: { ... } },
+    // ...
+  ];
+}
+---
+<!-- Your HTML template here. -->
+```
+
+The `getStaticPaths()` function should return an array of objects to determine which paths will be pre-rendered by Astro.
+
+It can also be used in static file endpoints for [dynamic routing](/en/guides/endpoints/#params-and-dynamic-routing).
+
+:::tip
+When using TypeScript, use the [`GetStaticPaths`](/en/guides/typescript/#infer-getstaticpaths-types) type utility to ensure type-safe access of your `params` and `props`.
+:::
+
+:::caution
+The `getStaticPaths()` function executes in its own isolated scope once, before any page loads. Therefore you can't reference anything from its parent scope, other than file imports. The compiler will warn you if you break this requirement.
+:::
+
+### `params`
+
+The `params` key of every returned object tells Astro what routes to build. The returned `params` must map back to the dynamic parameters and rest parameters defined in your component filepath.
+`params` are encoded into the URL, so only strings are supported as values. The value for each `params` object must match the parameters used in the page name.
+
+For example, suppose that you have a page at `src/pages/posts/[id].astro`. If you export `getStaticPaths` from this page and return the following for paths:
+
+```astro
+---
+export async function getStaticPaths() {
+  return [
+    { params: { id: '1' } },
+    { params: { id: '2' } },
+    { params: { id: '3' } }
+  ];
+}
+
+const { id } = Astro.params;
+---
+<h1>{id}</h1>
+```
+
+Then Astro will statically generate `posts/1`, `posts/2`, and `posts/3` at build time.
+
+### Data Passing with `props`
+
+To pass additional data to each generated page, you can also set a `props` value on every returned path object. Unlike `params`, `props` are not encoded into the URL and so aren't limited to only strings.
```

Comment:
This would align this with my `params` suggestion above:

```suggestion
To pass additional data to each generated page, you can set a `props` value on each object in the array returned by `getStaticPaths()`. Unlike `params`, `props` are not encoded into the URL and so aren't limited to only strings.
```